### PR TITLE
Pickle data only if necessary

### LINKED ISSUE #4: Borrow some ideas from django-nonrel/mongodb-cache
Just wanted to discuss a few ideas that could be borrowed from [mongodb-cache](https://github.com/django-nonrel/mongodb-cache):

- Pickle data only if needed like [here](https://github.com/django-nonrel/mongodb-cache/blob/19f28f4ce868462eae9019516cf8d8a6cbb0f7d8/django_mongodb_cache/backend.py#L69-L78)
- Raw data will allow for atomic increment like [here](https://github.com/django-nonrel/mongodb-cache/blob/19f28f4ce868462eae9019516cf8d8a6cbb0f7d8/django_mongodb_cache/backend.py#L82-L100)

Please let me know what you think.

### PR DESCRIPTION
**Pull request by romanukes**
Fix #4

- Try to save raw data and pickle if failed.
- Update `incr` method with `find_one_and_update` pymongo method. Potentially breaking change, only works with raw data.

Needs to be tested.

### CODE REVIEW

**Review by kjpc-tech — CHANGES_REQUESTED** [review]
I have something similar going right now. I think we need to keep `data` as the encoded version to not break this for existing users. And add the raw version as a new field perhaps `data_raw`. I don't know how this will work with the `incr` method as I'm not using that.

File: django_cache_with_mongodb/__init__.py [hunk 1]
```diff
@@ -218,9 +239,12 @@ def get_many(self, keys, version=None):
             }
         )
         for result in data:
-            unencoded = base64.decodebytes(result["data"])
-            unpickled = pickle.loads(unencoded)
-            out[parsed_keys[result["key"]]] = unpickled
+            if "encoded_data" in result:
```
Comment (kjpc-tech) [inline django_cache_with_mongodb/__init__.py]: This should be `data`

**Review by kjpc-tech — COMMENTED** [review]
One little comment. Other then that this seems good to me. The `incr` method may break existing users?

File: django_cache_with_mongodb/__init__.py (hunk 1 above)
Comment (romanukes) [reply django_cache_with_mongodb/__init__.py]: The `incr` method probably will break for the existing non-expired records in the cache. However, the current implementation should be faster and threadsafe. It should be a major version bump with a disclaimer.

**Comment by kjpc-tech** [comment]
I've done some light testing of this and its working great! I have not tested `incr` as its not something needed for my application. I hope this can get merged. Thanks!

**Comment by Olivier-OH** [comment]
I'll merge as soon as possible. Thanks for your contribution.

**Comment by Olivier-OH** [comment]
Merged and released on PyPI :-)

**Pull request merged**